algorithms/src/sorting/impl/Kokkos_SortByKeyImpl.hpp

```diff
@@ -142,8 +142,10 @@ void sort_by_key_rocthrust(
 
 #if defined(KOKKOS_ENABLE_ONEDPL)
 
-#if ONEDPL_VERSION_MAJOR > 2022 || \
-    (ONEDPL_VERSION_MAJOR == 2022 && ONEDPL_VERSION_MINOR > 7)
+#if ONEDPL_VERSION_MAJOR > 2022 ||   \
+    (ONEDPL_VERSION_MAJOR == 2022 && \
+     (ONEDPL_VERSION_MINOR > 7 ||    \
+      (ONEDPL_VERSION_MINOR == 7 && ONEDPL_VERSION_PATCH >= 1)))
```

Comment:
This is so ugly that it makes me think we might want to define some semantic versioning comparison macros.
(Obviously not blocking and would need to be done elsewhere)

In the meantime consider defining a `KOKKOS_IMPL_SYCL_ALWAYS_USE_ONEDPL_SORTING` or whatnot